server.go

```diff
@@ -218,8 +221,15 @@ func handlePacket(s *Server, p orderedRequest) error {
 			rpkt = statusFromError(p.ID, err)
 		}
 	case *sshFxpMkdirPacket:
-		// TODO FIXME: ignore flags field
-		err := os.Mkdir(s.toLocalPath(p.Path), 0o755)
+		var mode os.FileMode = defaultDirMode
+		if p.Attrs != nil {
+			attrs, _ := unmarshalFileStat(p.Flags, p.Attrs.([]byte))
```

Comment:
Either `p.Attrs` should be typed `[]byte` or we should properly handle the case where it is not `[]byte`.

It should be sufficient to use:
```
switch pattrs := p.Attrs.(type) {
case []byte:
  …
case *FileStat:
  … // since this would be a fast toFileMode(pattrs.Mode) might as well support it.
}
```

(I can’t speak to the legacy code doing differently. If I were redoing the FXP_SET_STAT packets, I’d do it quite a bit differently… I mean, I already did it’s in `internal/encoding/ssh/filexfer`…)